iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -469,11 +462,19 @@ private async Task EnsureOpenedAsync(CancellationToken cancellationToken)
                 {
                     if (Logging.IsEnabled) Logging.Info(this, "Opening connection", nameof(EnsureOpenedAsync));
                     await OpenAsyncInternal(cancellationToken).ConfigureAwait(false);
-                    _opened = true;
-                    _openCalled = true;
+                    if (!_disposed)
+                    {
+                        _opened = true;
+                        _openCalled = true;
 
-                    // Send the request for transport close notification.
-                    _transportClosedTask = HandleDisconnect();
+                        // Send the request for transport close notification.
+                        _transportClosedTask = HandleDisconnect();
+                    }
+                    else
+                    {
+                        if (Logging.IsEnabled) Logging.Info(this, "Race condition: Disposed during opening.", nameof(EnsureOpenedAsync));
```

Comment:
A race condition is a bug. We should definitely not log that we have a bug: either solve it or file an issue and add a TODO item.

Reply:
Race condition is not a bug. It will happen. Here just state it happened. It should be handled already for AMQP in `AmqpUnit`.

Reply:
If logging is not the right way to do it, then can you please propose a way to address  this?

Reply:
Table P0

Reply:
See the design spec: this needs to be handled by the transport handle.
Also the log shouldn’t say “race condition”.